lib/model.js

```diff
@@ -3399,7 +3398,13 @@ Model.bulkWrite = async function bulkWrite(ops, options) {
   const ordered = options.ordered == null ? true : options.ordered;
 
   if (ops.length === 0) {
-    return getDefaultBulkwriteResult();
+    const BulkWriteResult = this.base.driver.get().BulkWriteResult;
+    const bulkWriteResult = new BulkWriteResult(getDefaultBulkwriteResult(), false);
+    bulkWriteResult.n = 0;
+    bulkWriteResult.mongoose = bulkWriteResult.mongoose || {};
+    bulkWriteResult.mongoose.validationErrors = [];
+    bulkWriteResult.mongoose.results = [];
```

Comment:
This looks like a somewhat common operation, should this maybe be its own function?

Reply:
Done in 633ea9d